server/src/main/java/org/elasticsearch/common/logging/LoggingOutputStream.java

```diff
@@ -91,7 +91,12 @@ public void write(int b) throws IOException {
     public void flush() {
         Buffer buffer = threadLocal.get();
         if (buffer.used == 0) return;
-        log(new String(buffer.bytes, 0, buffer.used, StandardCharsets.UTF_8));
+        int used = buffer.used;
+        if (buffer.bytes[used - 1] == '\r') {
```

Comment:
Wouldn't this throw a `ArrayIndexOutOfBounds` exception for instances where we attempt to write a line which is simply a single `\n` character? I think we need to add a bounds check here.

Reply:
`used - 1` is the last character of the buffer, and we know the buffer size is at least 1 because of the check on line 93.

Reply:
I see what's happening now. I misunderstood the `\n` character itself to be part of this buffer but it's not. Nevermind.